File: endpoint/models/student.go
package models

type Student struct {
	ID         uint    `json:"id" gorm:"primaryKey"`
	Username   string  `json:"username" gorm:"size:255;not null"`
	Email      string  `json:"email" gorm:"not null;unique"`
	Password   string  `json:"-" gorm:"not null"`
	Nrp        *string `json:"nrp"`
	MaxCredits *int    `json:"max_credits"`

	BaseModel
}

File: frontend/app/models/student.ts
import type { BaseId } from "./baseId";
import type { BaseTimestamp } from "./baseTimestamp";
import type { CoursePlan } from "./coursePlan";

export interface Student extends BaseId, BaseTimestamp {
    username: string;
    email?: string;
    password?: string | null;
    nrp: string;
    maxCredits?: number;
    coursePlans?: CoursePlan[];
}
